### PR TITLE
Multiple build environments

### PR DESCRIPTION
**Pull request by itziakos**
This PRs adds support for multiple build environments.  The test coverage changes are not expected to change since there are not many gui toolkit specific tests

### CODE REVIEW

**Comment by itziakos** [comment]
Added support for pillow. There is a strange problem with nose and expectedFailure decorated methods. Need to investigate.

**Comment by itziakos** [comment]
This PR is ready for review.

File: travis_setup.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Simlinks for PIL compilation
+sudo ln -s /usr/lib/`uname -i`-linux-gnu/libfreetype.so /usr/lib/
+sudo ln -s /usr/lib/`uname -i`-linux-gnu/libjpeg.so /usr/lib/
+sudo ln -s /usr/lib/`uname -i`-linux-gnu/libpng.so /usr/lib/
+sudo ln -s /usr/lib/`uname -i`-linux-gnu/libz.so /usr/lib/
+
+# setup gui
+if [ "$ETS_TOOLKIT" = "wx" ]; then
+    echo "HERE"
```
Comment (jwiggins) [inline travis_setup.sh]: I'm guessing this isn't needed any longer?

File: travis_setup.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+fi
+
+# compile cairo
+wget -nv http://cairographics.org/releases/py2cairo-1.10.0.tar.bz2
+tar -xf py2cairo-1.10.0.tar.bz2
+cd py2cairo-1.10.0
+./waf configure
+./waf build
+sudo ./waf install
+cd ..
```
Comment (jwiggins) [inline travis_setup.sh]: Use `pushd`/`popd` instead of `cd`

**Comment by jwiggins** [comment]
Any reason why the setup script can't be named with a leading `.`? That's how the install scripts were named before. I noticed several commits where you kinda bounced around trying to find something that Travis would accept.

Other than some pretty minor feedback, I'm :+1: on this. Hopefully this will be a bit more stable than the earlier accommodations made for PIL/pillow. Oh, and actually testing WX is a nice touch too :wink:

**Comment by itziakos** [comment]
> Any reason why the setup script can't be named with a leading .? That's how the install scripts were named before. I noticed several commits where you kinda bounced around trying to find something that Travis would accept.

it looked to me weird that the scripts where hidden, so I changed that.

**Comment by itziakos** [comment]
> Other than some pretty minor feedback, I'm :+1: on this. Hopefully this will be a bit more stable than the earlier accommodations made for PIL/pillow. Oh, and actually testing WX is a nice touch too :wink:

In general, I think it is a better build setup. Please note however that it does not look that we are testing much or WX or QT code of enable. We need to increase the test coverage.

**Comment by jwiggins** [comment]
WTF. `pushd`/`popd` doesn't work when you execute a script with `sh`? Perhaps `/bin/sh` is not bash on Travis?

**Comment by WarrenWeckesser** [comment]
Ubuntu uses `dash` for `/bin/sh`, and `dash` does not has `pushd` and `popd`.

**Comment by itziakos** [comment]
Ok, I reverted back, it is not that bad to use `cd`

**Comment by jwiggins** [comment]
Thanks @itziakos! :shipit:

**Pull request merged**